core/src/main/java/com/google/errorprone/bugpatterns/UnsafeLocaleUsage.java

```diff
@@ -54,11 +57,27 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
           .setMessage(
               "Avoid using Locale.toString() since it produces a value that"
                   + " misleadingly looks like a locale identifier. Prefer using"
-                  + " Locale.toLanguageTag() since it produces an IETF BCP 47-formatted string that"
-                  + " can be deserialized back into a Locale.")
+                  + " Locale.toLanguageTag() since it produces an IETF BCP 47-formatted string"
+                  + " that can be deserialized back into a Locale.")
           .addFix(SuggestedFixes.renameMethodInvocation(tree, "toLanguageTag", state))
           .build();
     }
+    if (LOCALE_OF.matches(tree, state)) {
+      Description.Builder descriptionBuilder =
+          buildDescription(tree)
+              .setMessage(
+                  "Avoid using Locale.of static methods (which do not check their arguments for"
```

Comment:
By default, the message is the `BugPattern.summary`. Is there information here that applies to all findings that could go in the summary instead?

If you want to customize this to mention e.g. `Locale.of` vs. `Locale constructors`, WDYT about refactoring and using a format string to share the text that's the same for both?

Reply:
Done